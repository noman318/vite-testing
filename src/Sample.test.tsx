import { render, screen, within } from "@testing-library/react";
import Sample from "./Sample";

describe("Post Component", () => {
  it("renders with no likes", () => {
    render(<Sample content={"it's good"} user={"Noman"} />);
    // screen.debug();
    const postContainer = screen.getByTestId("post-container");
    expect(postContainer).toBeInTheDocument();

    const user = screen.getByRole("heading");
    expect(user).toBeInTheDocument();
    expect(user).toHaveTextContent("Noman");

    const para = screen.getByRole("paragraph");
    expect(para).toBeInTheDocument();
    expect(para).toHaveTextContent("it's good");

    const likes = screen.queryByRole("list");
    expect(likes).not.toBeInTheDocument();
  });

  it("renders with likes", () => {
    render(
      <Sample
        content={"it's good"}
        user={"Noman"}
        likesBy={["Noman", "Aman", "etcs"]}
      />
    );
    const likesScreen = screen.getByTestId("likes-container");

    // expect(likesScreen).toBeInTheDocument();

    const likes = within(likesScreen).getAllByRole("listitem");
    expect(likes).toHaveLength(3);
    expect(likes[0]).toHaveTextContent("Noman");
    expect(likes[1]).toHaveTextContent("Aman");
    expect(likes[2]).toHaveTextContent("etcs");
  });
});
