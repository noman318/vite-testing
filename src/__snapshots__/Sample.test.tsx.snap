// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post Component > snapshot test 1`] = `
<DocumentFragment>
  <div
    data-testid="post-container"
  >
    Add commentMore actions
    <h2>
      Noman:
    </h2>
    <p>
      it's good
    </p>
    <div
      data-testid="likes-container"
    >
      <h3>
        Likes by:
      </h3>
      <ul>
        <li>
          Noman
        </li>
        <li>
          Aman
        </li>
        <li>
          etcs
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
